app/controllers/chat_controller.rb

```diff
@@ -344,11 +344,11 @@ def move_messages_to_channel
     params.require(:message_ids)
     params.require(:destination_channel_id)
 
-    raise Discourse::InvalidAccess if !guardian.can_move_chat_messages?
-
     chat_channel = DiscourseChat::ChatChannelFetcher.find_with_access_check(params[:chat_channel_id], guardian)
     destination_channel = DiscourseChat::ChatChannelFetcher.find_with_access_check(params[:destination_channel_id], guardian)
 
+    raise Discourse::InvalidAccess if !guardian.can_move_chat_messages?(chat_channel)
```

Comment:
I would move this just right after `chat_channel = ....`. That way, we don't fetch `destination_channel` for nothing if the guardian check fails.